cli/src/main/java/com/devonfw/tools/ide/url/updater/AbstractUrlUpdater.java

```diff
@@ -303,28 +315,23 @@ protected boolean isSuccess(HttpResponse<?> response) {
   /**
    * Checks if the download file checksum is still valid
    *
-   * @param url String of the URL to check
-   * @param urlVersion the {@link UrlVersion} with the {@link UrlVersion#getName() version-number} to process.
-   * @param os the {@link OperatingSystem}
-   * @param architecture the {@link SystemArchitecture}
-   * @param checksum String of the new checksum to check
-   * @param tool String of the tool
-   * @param version String of the version
+   * @param checksum the newly computed checksum.
+   * @param urlChecksum the {@link UrlChecksum} to compare.
+   * @param toolWithEdition the tool/edition information for logging.
+   * @param version the tool version.
+   * @param url the URL the checksum belongs to.
    * @return {@code true} if update of checksum was successful, {@code false} otherwise.
    */
-  private static boolean isChecksumStillValid(String url, UrlVersion urlVersion, OperatingSystem os,
-      SystemArchitecture architecture, String checksum, String tool, String version) {
+  private static boolean isChecksumStillValid(String checksum, UrlChecksum urlChecksum,
+      String toolWithEdition, String version, String url) {
 
-    UrlDownloadFile urlDownloadFile = urlVersion.getOrCreateUrls(os, architecture);
-    UrlChecksum urlChecksum = urlVersion.getOrCreateChecksum(urlDownloadFile.getName());
-    String oldChecksum = urlChecksum.getChecksum();
+    String existingChecksum = urlChecksum.getChecksum();
 
-    if ((oldChecksum != null) && !Objects.equals(oldChecksum, checksum)) {
-      logger.error("For tool {} and version {} the mirror URL {} points to a different checksum {} but expected {}.",
-          tool, version, url, checksum, oldChecksum);
+    if (!Objects.equals(existingChecksum, checksum)) {
+      logger.error("For tool {} and version {} the download URL {} results in checksum {} but expected {}.",
+          toolWithEdition, version, url, checksum, existingChecksum);
       return false;
     } else {
-      urlDownloadFile.addUrl(url);
```

Comment:
bad design: the method name `isChecksumStillValid` implies that we only check the checksum here and not do such side-effects.